### PR TITLE
Optimization of delete and insert

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo <xinguo.li@zilliz.com>

### CODE REVIEW

File: internal/proxy/util.go [hunk 1]
```diff
@@ -878,3 +884,223 @@ func isPartitionLoaded(ctx context.Context, qc types.QueryCoord, collID int64, p
...
+	ids, err := parsePrimaryFieldData2IDs(primaryFieldData)
+	if err != nil {
+		log.Error("parse primary field data to IDs failed", zap.String("collectionName", insertMsg.CollectionName), zap.Error(err))
+		return nil, err
+	}
+
+	return ids, nil
+}
+
+func assignSegmentID(ctx context.Context, insertMsg *msgstream.InsertMsg, result *milvuspb.MutationResult, channelNames []string, idAllocator *allocator.IDAllocator, segIDAssigner *segIDAssigner) (*msgstream.MsgPack, error) {
```
Comment (jiaoew1991) [inline internal/proxy/util.go:954]: move it to a separate file, it is too long and complicated to be a util function

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**